### PR TITLE
Draft: Doc for the 2017 session

### PR DESCRIPTION
**Pull request by uggla**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,14 @@
 Cloud Native Application
 ========================
 
-This document purpose is to describe the Cloud Native Application training described below and which is in particular delivered to ENSIMAG IS students in 2016 and 2017.
+This document purpose is to describe the Cloud Native Application training described below and which is in particular delivered to ENSIMAG IS students in 2017 and 2018.
```
Comment (bcornec) [inline README.md:4]: 2 times describe in the original doc was also too much...
What I meant was in 2016-2017 and 2017-2018.

File: README.md [hunk 2]
```diff
@@ -23,92 +24,100 @@ The main OpenStack entry point is at http://www.openstack.org
...
+## Goals & constraints
 
-Practitioners will first have to automate the management of a virtual infrastructure (Infrastructure as code concept). Within this part we are using Openstack as our Infrastructure as a Service platform (IaaS) but this is an implementation choice. The same principles can be implemented on top of another IaaS platform like Eucalyptus, Cloudstack, Amazon AWS, Azure, Docker. The teachers team will provide two physical infrastructures for that purpose, one which will be pre-installed with an OpenStack distribution in which a tenant per group of student will be available for application development, and another one (set of 15 servers) which will be pre-installed with just a CentOS 7 Operating System. Students will configure the IaaS part on top of it to study the various components that they want to test. 
+### Goals
+As a company, the idea is to leverage an open source cloud native application to use it internally. Because the application is a cloud native one and will have to be frequently improved, the choice is to use the devops methodology to maintain this application into production.
```
Comment (bcornec) [inline README.md]: Already made comment to Christophe patch around methodology...

File: README.md [hunk 3]
```diff
@@ -23,92 +24,100 @@ The main OpenStack entry point is at http://www.openstack.org
...
 
-This part will be **evaluated on 4 points**.
+As a consequence, practitioners will first have to:
+ * Organize the team to work using devops best practices.
+ * Automate the management of a virtual infrastructure (Infrastructure as code concept). Within this part we are using Openstack as our Infrastructure as a Service platform (IaaS) but this is an implementation choice. The same principles can be implemented on top of another IaaS platform like Eucalyptus, Cloudstack, Amazon AWS, Azure, Docker.
+ * Build production environment, the teachers team will provide two physical infrastructures for that purpose, one which will be pre-installed with an OpenStack distribution in which a tenant per group of student will be available for application development, and another one (set of 15 servers) which will be pre-installed with just a CentOS 7 Operating System. Students will configure the IaaS part on top of it to study the various components that they want to test.
+ * Define and implement a CICD pipeline to easily test and deploy the application to staging area then production.
```
Comment (bcornec) [inline README.md]: CI/CD
in a staging area and then in production

File: README.md [hunk 4]
```diff
@@ -23,92 +24,100 @@ The main OpenStack entry point is at http://www.openstack.org
...
+ * Put in place tooling to share/track application changes beetween team members/customers.
+ * Ensure correct level of security within the factory. (no private keys or data available publicly, user access restriction, ports filtering...)
```
Comment (bcornec) [inline README.md]: Ensure a correct...

File: README.md [hunk 5]
```diff
@@ -23,92 +24,100 @@ The main OpenStack entry point is at http://www.openstack.org
...
- * P(icture) service: Look into Swift with customer ID to display the image of the customer, empty if no image.
+### Constraints
+ * Application should only expose http[s] (ports 80 and 443) to the external network.
+ * All materials should be kept on public git repository (github e.g.) with an Open Source license (Cf: https://opensource.org/licenses prefer the popular ones). Please note that your github history should reflect changes to the factory/infrastructure/application. Also, you should commit properly with a your personal credentials (email preferred) with a unique account.
+ * There is no restriction regarding the tools to implement the pipeline. It could be external services and/or on premise tools. However, be prepared to justify some choices.
 
-This part will be **evaluated on 6 points**.
+## Bonus goals
+ * Put in place monitoring to ensure application in working as expected.
```
Comment (bcornec) [inline README.md]: ensure that the application is working...

File: README.md [hunk 6]
```diff
@@ -23,92 +24,100 @@ The main OpenStack entry point is at http://www.openstack.org
...
+ * Put in place name resolution.
+ * Improve applications.
+ * Optimize testing duration.
+ * Communication dashboard  about application health to users.
+ * Performance of the application improvement by scaling services.
```
Comment (bcornec) [inline README.md]: Improvement of the application performances ...

File: README.md [hunk 7]
```diff
@@ -23,92 +24,100 @@ The main OpenStack entry point is at http://www.openstack.org
...
+ * Application reliability.
 
-The content provided by the teachers team will be referenced on the etherpad used to follow the exchanges on the project (https://etherpad.openstack.org/p/ENSIMAG_2017)
```
Comment (bcornec) [inline README.md]: Maybe we can keep the Etherpad link somewhere no ? (with a 2018 name of course)

File: README.md [hunk 8]
```diff
@@ -23,92 +24,100 @@ The main OpenStack entry point is at http://www.openstack.org
...
+# Application information
 
-## Goals & constraints
+## Application schema
+![applicaiton schema](schema/archi_docker.png)
```
Comment (bcornec) [inline README.md]: application

File: README.md [hunk 9]
```diff
@@ -23,92 +24,100 @@ The main OpenStack entry point is at http://www.openstack.org
...
 
 ## Teachers support
 
-Teachers will provide a remote access to the pre-production platform (set of servers preinstalled with a CentOS 7 distribution) that the IaaS team will have to setup from an OpenStack perspective and deliver so the DevOps teams could deploy their application on it. Teachers will also provide a production platform running an OpenStack distribution as a development environment for the application developers for testing purposes. 
-A mailing list has been created to provide remote support to students during the whole period. Various people will so be available to answer questions and help with regards to platform setup and access. The mailing-list address is ensimag-openstack@lists.osp.hpe.com 
+Teachers will provide a remote access to the production platform (set of servers preinstalled with a CentOS 7 distribution) that the team will have to setup from an OpenStack perspective and deliver teams could deploy their application on it. Teachers will also provide a pre-production platform running an OpenStack distribution as a development environment for the application developers for testing purposes.
```
Comment (bcornec) [inline README.md]: I thought we were calling it reference platform not pre-prod ;-)))

**Review by bcornec — CHANGES_REQUESTED** [review]

File: README.md (hunk 9 above)
Comment (clarsonneur) [reply README.md]: Correct. Look how I named it in my proposal... Trying to bring it in the story...

**Pull request merged**